src/steps/lead-campaign-id-equals.ts

```diff
@@ -0,0 +1,49 @@
+/*tslint:disable:no-else-after-return*/
+
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class LeadCampaignIdEquals extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Check Lead Campaign Membership';
+  /* tslint:disable-next-line:max-line-length */
+  protected stepExpression: string = 'the salesforce lead (?<email>.+) should be a member of campaign (?<campaignId>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'email',
+    type: FieldDefinition.Type.EMAIL,
+    description: "Lead's email address",
+  }, {
+    field: 'campaignId',
+    type: FieldDefinition.Type.STRING,
+    description: "Lead's CampaignId",
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const email: string = stepData.email;
+    const field: string = 'CampaignId';
+    const expectedValue: string = stepData.campaignId;
+    let lead: Record<string, any>;
```

Comment:
For the lead campaign id check i just changed all the 'lead' to 'campaignmembers' instead since lead doesnt even have a campaignId. The sobject is also campaignMember, so it would make sense